shuup/admin/utils/product.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+# This file is part of Shuup.
+#
+# Copyright (c) 2012-2021, Shoop Commerce Ltd. All rights reserved.
+#
+# This source code is licensed under the OSL-3.0 license found in the
+# LICENSE file in the root directory of this source tree.
+from django.http import HttpResponseRedirect
+from django.db.transaction import atomic
+from django.contrib import messages
+
+from shuup.core.models import (
+    Product, ProductAttribute, ProductMedia, ShopProduct
+)
+from shuup.admin.signals import product_copied
+from shuup.utils.models import get_data_dict
+
+class ProductCloner():
+    def __init__(self, current_shop, current_supplier):
+        self.current_shop = current_shop
+        self.current_supplier = current_supplier
+    
+    @atomic
+    def clone_product(self, shop_product=None):
+        product = shop_product.product
+        product_data = get_data_dict(product)
+        product_data.update({
+            "sku": "{}-{}".format(product.sku, Product.objects.count())
+        })
+        new_product = Product.objects.create(**product_data)
+        new_product.name = product.name
+        new_product.short_description = product.short_description
+        new_product.description = product.description
+        new_product.slug = product.slug
+        new_product.save()
+
+        new_shop_product = ShopProduct.objects.create(product=new_product, shop=self.current_shop)
```

Comment:
maybe use `shuup.utils.models.copy_model_instance` and update the product instance before saving. You also need to go through all the translated fields and copy them, just like in Product